core/templates/pages/library-page/search-bar/search-bar.component.ts

```diff
@@ -68,6 +68,9 @@ angular.module('oppia').component('searchBar', {
        */
       ctrl.openSubmenu = function(evt, menuName) {
         NavigationService.openSubmenu(evt, menuName);
+        // TODO(#8521): Remove the use of $rootScope.$apply()
+        // once the directive is migrated to angular.
+        $rootScope.$applyAsync();
```

Comment:
Here and below: why is this required now?

Reply:
@kevintab95 Navigation Service was migrated with this PR and there was  change detection issue with search bar in the library page,this was causing an e2e test to fail and it was not working properly,therefore it was required.

Reply:
Just to understand a bit better, what change is not getting detected in this particular scenario? Is it that the submenu is not opening?

Reply:
@kevintab95 Yes the dropdowns + the loader that appears to the left of search-bar when we input something into it